manual/manual/tutorials/coreexamples.etex

```diff
@@ -580,6 +580,69 @@ let fixpoint f x =
 the function "f" cannot raise a "Done" exception, which removes an
 entire class of misbehaving functions.
 
+\section{Lazy expressions}
+
+OCaml allows to defer some computation, usually expensive one, until later when 
+we need that expression evaluated. The deferred expression has type "'a Lazy.t", 
+where "'a" is the type of the evaluated expression. 
+
+We use "lazy (expr)" to delay the evaluation of some expression "expr". For 
+example, we can defer the computation of "1+1" until we need the result of that
+expression, i.e. "2". 
+
+\begin{caml_example}{toplevel}
+let lazy_two = lazy (1+1);;
+\end{caml_example}
+
+Note: "'a lazy_t" is a type constructor for keyword "lazy" used by the compiler. 
+One should use type "'a Lazy.t" to work with lazy expressions. 
+
+Also note that the value of "lazy_two" is "<lazy>", which means 
+the expression is not evaluated yet and the result is unknown.
+
+When we finally need the result of the expression, we can call "Lazy.force"  
+on the lazy expression to force the evaluation of the lazy expression. "force" 
+is a function from a standard-library  
+\href{https://caml.inria.fr/pub/docs/manual-ocaml/libref/Lazy.html}
+{module "Lazy"}.
+
+\begin{caml_example}{toplevel}
+  Lazy.force lazy_two;;
+\end{caml_example}
+
+Pay attention that our function call returns plain, unboxed value of the 
+expression. 
+
+We should also note that `Lazy.force` memoizes the result of forced computation. 
+In other words, every subsequent call of "Lazy.force" on the lazy expression is 
+constant time regardless how expensive the initial computation is.
```

Comment:
It may be worth it to be a bit more plain here (otherwise, constant time is both technical and not that clear) . Maybe something like

> In other words, every subsequent call of "Lazy.force" returns the result of the first computation
without recomputing the lazy expression.

?

Reply:
Your version seems easier to understand 👍